### PR TITLE
Show code coverage not supported message on non windows

### PR DESCRIPTION
**Pull request by smadala**
## Description
- Display proper warning when user try to collect code coverage on Unix.

**Current warning**
```
Data collection : Data collector 'Code Coverage' threw an exception during type loading, construction, or initialization: System.Reflection.TargetInvocationException:Exception has been thrown by the target of an invocation. ---> System.DllNotFoundException: Unable to load shared library 'kernel32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libkernel32.dll: cannot open shared object file: No such file or directory
   at Microsoft.VisualStudio.TraceCollector.ProcessJobObject.WinAPI.CreateJobObject(IntPtr jobAttributes, String name)
   at Microsoft.VisualStudio.TraceCollector.ProcessJobObject.CreateJobObject()
   at Microsoft.VisualStudio.Coverage.Vanguard..ctor()
   at Microsoft.VisualStudio.Coverage.DynamicCoverageDataCollectorImpl..ctor()
   at Microsoft.VisualStudio.Coverage.DynamicCoverageDataCollector..ctor()
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.TestPluginManager.CreateTestExtension[T](Type extensionType)
   at Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.Utilities.LazyExtension`2.get_Value()
   at Microsoft.VisualStudio.TestPlatform.Common.DataCollector.DataCollectionManager.TryGetTestExtension(String extensionUri)
   at Microsoft.VisualStudio.TestPlatform.Common.DataCollector.DataCollectionManager.LoadAndInitialize(DataCollectorSettings dataCollectorSettings, String settingsXml).
```

### CODE REVIEW

File: src/DataCollectors/TraceDataCollector/Resources/Resources.resx [hunk 1]
```diff
@@ -132,4 +132,7 @@
   <data name="VanguardConnectionTimeout" xml:space="preserve">
     <value>Failed to receive running event from CodeCoverage.exe in {0} seconds, This may occur due to machine slowness, please set environment variable {1} to increase timeout.</value>
   </data>
+  <data name="CodeCoverageOnlySupportsWindows" xml:space="preserve">
+    <value>Currently code coverage support available only for Windows operating system. No code coverage data available for current test run.</value>
```
Comment (mayankbansal018) [inline src/DataCollectors/TraceDataCollector/Resources/Resources.resx]: Get this verified by Pratap

File: src/DataCollectors/TraceDataCollector/VanguardCollector/DynamicCoverageDataCollector.cs [hunk 2]
```diff
@@ -10,8 +10,10 @@ namespace Microsoft.VisualStudio.Coverage
     using System.Xml;
     using Interfaces;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
```
Comment (mayankbansal018) [inline src/DataCollectors/TraceDataCollector/VanguardCollector/DynamicCoverageDataCollector.cs]: I believe platform abstraction is core part of TP, & any extension should not take a dependency on it.

**Review by singhsarab — APPROVED** [review]

File: src/DataCollectors/TraceDataCollector/Resources/Resources.resx (hunk 1 above)
Comment (pvlakshm) [reply src/DataCollectors/TraceDataCollector/Resources/Resources.resx]: How about: "No code coverage data available. Code coverage is currently supported only on Windows."

Comment (smadala) [reply src/DataCollectors/TraceDataCollector/Resources/Resources.resx]: @pvlakshm sounds good. Thanks.

File: src/DataCollectors/TraceDataCollector/VanguardCollector/DynamicCoverageDataCollector.cs (hunk 2 above)
Comment (smadala) [reply src/DataCollectors/TraceDataCollector/VanguardCollector/DynamicCoverageDataCollector.cs]: Added code in TDC for getting current OS name.

**Review by mayankbansal018 — APPROVED** [review]

**Pull request merged**